### PR TITLE
PR:  Upgrade HTTP links to HTTPS and update links targets for redirects/broken links

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
Similar to spyder-ide/spyder#7611 , upgrades HTTP links to HTTPS where available to conform to modern best practices (and perhaps slightly help our pagerank, if certain rumors are to be believed), and fixes the link target if the link is broken, redirected or out of date. All links were hand-tested before and after conversion to ensure they loaded properly, and links that were functional in nature (there were none detected) and URNs/URIs as part of HTML/XML schema were not modified.

### CODE REVIEW

File: assets/static/js/app.js [hunk 1]
```diff
@@ -1,23 +1,23 @@
 !function(e){function t(i){if(n[i])return n[i].exports;var o=n[i]={exports:{},id:i,loaded:!1};return e[i].call(o.exports,o,o.exports,t),o.loaded=!0,o.exports}var n={};return t.m=e,t.c=n,t.p="/static",t(0)}([function(e,t,n){(function(e){foundation=n(2),n(3),n(14),e(document).ready(function(e){e(".carousel-default").slick({}),e(".carousel-home").slick({autoplay:!0,arrows:!1,autoplaySpeed:2e3}),e(".carousel-users").slick({autoplay:!0,infinite:!0,arrows:!0,slidesToShow:3,slidesToScroll:3,autoplaySpeed:5e3,responsive:[{breakpoint:1024,settings:{slidesToShow:2,slidesToScroll:2}},{breakpoint:480,settings:{slidesToShow:1,slidesToScroll:1}}]}),e(document).foundation()})}).call(t,n(1))},function(e,t,n){var i,o;/*!
```
Comment (ccordoba12) [inline assets/static/js/app.js:1]: Please revert this change. It makes no sense at all to touch Javascript asset files in this PR.

Comment (CAM-Gerlach) [reply assets/static/js/app.js:1]: Sorry, that was accidentally still included due to an oversight on my part. I had modified one "link" in that file and then saved the file with autostrip trailing spaces turned on, then quickly realized it how silly I was that it was an XML namespace URN, not an actual URL, and undid the change before I pushed (I also checked my other HTTPS PRs to make sure it didn't do something silly like this). However, I missed reverting the whitespace cleanup. Properly fixed now.

**Comment by CAM-Gerlach** [comment]
@ccordoba12 Dropped silly errant commit; thanks for catching that.

**Review by ccordoba12 — APPROVED** [review]

**Comment by ccordoba12** [comment]
This is in `production`. Thanks @CAM-Gerlach!

**Comment by CAM-Gerlach** [comment]
Thanks @ccordoba12 !
